demux/cue.c

```diff
@@ -214,6 +214,8 @@ struct cue_file *mp_parse_cue(struct bstr data)
             talloc_free(f);
             return NULL;
         case CUE_TRACK: {
+            if (bstr_find0(param, "AUDIO") == -1)
```

Comment:
Is this certain that `AUDIO` part will be there? Wouldn't be safer to reject `BINARY` for example? Also does it have to be `find`? Isn't this we don't what the full string is to tokenize it correctly?

Reply:
optional according to https://wyday.com/cuesharp/specification.php

why find? the line could look like any of these:
> TRACK 03 "AUDIO"
> TRACK 03 'AUDIO'

(incidentally we don't handle single quotes)

Reply:
Indicating the track mode is required but we don't support anything other than `AUDIO`. So indeed, it might not be there, but if it is not, then it is something we shouldn't support. I don't think quotes or any of that other stuff valid cuespec so as long as `AUDIO` appears it should be OK.

Reply:
I checked 412 cue files I have lying around and they all satisfy ` *TRACK [0-9]+ AUDIO$`.

`xargs -a <(find . -name '*.cue') -d $'\n' grep -Eh '\bTRACK\b' | sort -u` in case anyone wants to do the same

Reply:
We can just do `bstr_endswith0` in that case.

Reply:
why? if we're worrying about spaces it should be `bstr_find0` and if not `bstr_equals0`

Reply:
I thought we just decided to not worry about trailing spaces, quotes, etc.

Reply:
But if there are quotes endswith would fail, no?

Reply:
Well yes, but I've never seen a cue file that is not in the form of `TRACK [0-9]+ [SOMETYPE]`. It doesn't really matter either way to me though. We can use `bstr_find0` if you guys really want to support people with likely broken files.

Reply:
I vote for bstr_find0 then

Reply:
Changed it back to `bstr_find0`.

Reply:
Fine for me.